packages/wrangler/src/deployment-bundle/create-worker-upload-form.ts

```diff
@@ -171,14 +172,14 @@ export function createWorkerUploadForm(worker: CfWorkerInit): FormData {
 		tail_consumers,
 		limits,
 		annotations,
-		experimental_assets,
+		experimental_assets_jwt,
 	} = worker;
 
 	// short circuit if static assets upload only
-	if (experimental_assets?.staticAssetsOnly) {
+	if (main.name === "no-op-assets-worker.js") {
```

Comment:
this makes me a bit nervous, but I can't think of a better solution rn. I also feel the way we will solve https://github.com/cloudflare/workers-sdk/pull/6474#discussion_r1716794688 will highly influence how we want to perform this detection

Reply:
Actually I should've put it behind the experimental assets flag as well - that should make it a bit more safe for now til/if we come up with a better solution

Reply:
raised in WC-2534